Lib/uuid.py

```diff
@@ -719,14 +720,72 @@ def uuid5(namespace, name):
     hash = sha1(namespace.bytes + name).digest()
     return UUID(bytes=hash[:16], version=5)
 
+_last_timestamp_v7 = None
+_last_counter_v7 = 0  # 42-bit counter
+
+def uuid7():
+    """Generate a UUID from a Unix timestamp in milliseconds and random bits.
+
+    UUIDv7 objects feature monotonicity within a millisecond.
+    """
+    # --- 48 ---   -- 4 --   --- 12 ---   -- 2 --   --- 30 ---   - 32 -
+    # unix_ts_ms | version | counter_hi | variant | counter_lo | random
+    #
+    # 'counter = counter_hi | counter_lo' is a 42-bit counter constructed
+    # with Method 1 of RFC 9562, §6.2, and its MSB is set to 0.
+    #
+    # 'random' is a 32-bit random value regenerated for every new UUID.
+    #
+    # If multiple UUIDs are generated within the same millisecond, the LSB
+    # of 'counter' is incremented by 1. When overflowing, the timestamp is
+    # advanced and the counter is reset to a random 42-bit integer with MSB
+    # set to 0.
+
+    def get_counter_and_tail():
+        rand = int.from_bytes(os.urandom(10))
+        # 42-bit counter with MSB set to 0
+        counter = (rand >> 32) & 0x1ffffffffff
+        # 32-bit random data
+        tail = rand & 0xffffffff
```

Comment:
Small nit: `counter` and `tail` shadow local variables outside of this function. It'd be good to rename for clarify to avoid the shadow, unless it's somehow deliberate?

Reply:
It's deliberate and it's not really an issue. As you can see, it's used as `counter, tail = get_counter_and_tail()`.